src/openforms/contrib/customer_interactions/utils.py

```diff
@@ -0,0 +1,165 @@
+        sorted_addresses, key=lambda x: x["soortDigitaalAdres"]
+    )
+
+    result: list[CommunicationChannel] = []
+    for address_type, group_iter in grouped_digital_addresses:
+        group = list(group_iter)
+        channel_name: SupportedChannels = ADDRESS_TYPES_TO_CHANNELS[address_type]
+        if channel_name not in configured_address_types:
+            continue
+
+        group_preferences: CommunicationChannel = {
+            "type": channel_name,
+            "options": [address["adres"] for address in group],
+            "preferred": next(
+                (address["adres"] for address in group if address["isStandaardAdres"]),
+                None,
+            ),
+        }
+        result.append(group_preferences)
+    return result
+
+
+def update_customer_interaction_data(
+    submission: Submission, profile_key: str
+) -> dict[str, JSONValue] | None:
+    """
+    Writes to the Customer interaction API when the form with profile component is submitted
+
+    * always create contactMoment, betrokkene and onderwerpObject
+    * if the user is authenticated, it's always linked to partij
+    * if the address is marked as 'isNewPreferred' then it's linked to the partij
+      otherwise it's linked just to betrokkene
+    """
+    # bsn
+    bsn = (
+        submission.auth_info.value
+        if (
+            submission.is_authenticated
+            and submission.auth_info.attribute == AuthAttribute.bsn
+        )
+        else None
+    )
+
+    # submission profile data
+    state = submission.load_submission_value_variables_state()
+    profile_submission_data: list[DigitalAddress] = state.get_data()[profile_key]
+
+    # prefill info
+    try:
+        prefill_form_variable = submission.form.formvariable_set.get(
+            source=FormVariableSources.user_defined,
+            prefill_plugin="communication_preferences",  # todo replace with plugin identifier
+            prefill_options__profile_form_variable=profile_key,
+        )
+    except FormVariable.DoesNotExist:
+        # todo warning, since the component configured shouldUpdateCustomerData = True
+        return
+
+    api_group_id = prefill_form_variable.prefill_options[
+        "customer_interactions_api_group"
+    ]
+    api_group = CustomerInteractionsAPIGroupConfig.objects.get(identifier=api_group_id)
+    prefill_value = state.variables[prefill_form_variable.key].value
+
+    channels_to_address_types: Mapping[SupportedChannels, SoortDigitaalAdres] = {
+        v: k for k, v in ADDRESS_TYPES_TO_CHANNELS.items()
+    }
+
+    result = {}
+    with get_customer_interactions_client(api_group) as client:
+        if bsn:
+            # link authenticated user to the party
+            party = client.find_party_for_bsn(bsn)
+            if not party:
+                party = client.create_party_for_bsn(bsn)
+                result["partij"] = party
+
+            party_uuid = party["uuid"]
```

Comment:
Should `partij` not be included if it already exists?

```suggestion
        if bsn:
            # link authenticated user to the party
            party = client.find_party_for_bsn(bsn)
            if not party:
                party = client.create_party_for_bsn(bsn)
            
            result["partij"] = party
            party_uuid = party["uuid"]
```

Or is `result` only the newly created/updated data?

Reply:
I thought to include only new data there. We can include the existing partij, but maybe using another name (like "existing_partij") ? 
Should I include it?